### PR TITLE
Start moving away from Event calls and move to central dispatch event handling

### PR DESCRIPTION
**Pull request by XVhc6A**
### Brief description of the changes:

The Event.add comfy framework is very convenient for one off things that you want to add to FBB.  Unfortunately, it is a little too easy to just throw anywhere, and it can create a spaghetti nest of loaded or pending event handlers/tasks.

If we move to a centralized dispatch of event calling:
- We can greatly simplify the way events are handled
- We will have much better visibility into exactly what is happening - and when.
- We will lower the barrier to entry to the code, make our codebase more similar to other factorio mods/scenarios (https://github.com/raiguard/Krastorio2/blob/master/control.lua)
- We will be able to fine tune our events and scheduling to a better degree.

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

File: maps/biter_battles_v2/functions.lua [hunk 1]
```diff
@@ -263,8 +263,6 @@ function Public.is_biter_area(position,noise_Enabled)
 end
 
 function Public.no_turret_creep(event)
-	local entity = event.created_entity
```
Comment (Ragnarok77-factorio) [inline maps/biter_battles_v2/functions.lua:266]: ...mp/currently-playing/maps/biter_battles_v2/functions.lua:268: attempt to index global 'entity' (a nil value) in ...io/temp/currently-playing/maps/biter_battles_v2/main.lua:66

When you build a turret , or use ghost then bots to make it and so on.


https://discord.com/channels/823696400797138974/1003021908796784800/1190225821873479700

**Review by Ragnarok77-factorio — CHANGES_REQUESTED** [review]
Needs bug to be fixed first

**Comment by Ragnarok77-factorio** [comment]
Changed milestone until the bug is fixed (check review).

**Comment by XVhc6A** [comment]
testing locally doesn't yield any errors now.  also cleaned up some other functions as well.

**Comment by XVhc6A** [comment]
going to close this in favor of a split version.

**Pull request closed without merging**